bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/BigtableSyncMapper.java

```diff
@@ -0,0 +1,187 @@
+      BigtableSyncTableAccessor.setSourceZkClusterConf(conf, targetZkClusterConf);
+      LOG.info(
+          "source conn temporarily set as target for initialization only: " + targetZkClusterConf);
+    }
+
+    // SyncTable initializes map task configuration and connections with source and target. A valid
+    // source and target are required for initialization. The connection is not used in the setup
+    // and only required for initialization.
+    super.setup(context);
+
+    // establish target connection to bigtable
+    if (!conf.onlyKeyExists(TARGET_BT_PROJECTID_CONF_KEY)) {
+      createBigtableTargetConnection(conf);
+    }
+
+    // establish source connection to bigtable
+    if (!conf.onlyKeyExists(SOURCE_BT_PROJECTID_CONF_KEY)) {
+      createBigtableSourceConnection(conf);
+    }
+  }
+
+  /**
+   * create source connection with Bigtable
+   *
+   * @param conf
+   * @throws IOException
+   */
+  private void createBigtableSourceConnection(Configuration conf) throws IOException {
+    // re-create connection with bigtable
+    closeConnection(BigtableSyncTableAccessor.getSourceConnection(this), "source");
+
+    // create source connection config
+    // inherit base config, but override connection based on job args
+    Configuration bigtableConf =
+        getBigtableConfiguration(
+            new Configuration(conf),
+            SOURCE_BT_PROJECTID_CONF_KEY,
+            SOURCE_BT_INSTANCE_CONF_KEY,
+            SOURCE_BT_APP_PROFILE_CONF_KEY,
+            "source");
+    Connection srcConn = BigtableSyncTableAccessor.setSourceConnection(this, bigtableConf);
+    BigtableSyncTableAccessor.setSourceTable(this, srcConn, bigtableConf);
+  }
+
+  /**
+   * create target connection with Bigtable
+   *
+   * @param conf
+   * @throws IOException
+   */
+  private void createBigtableTargetConnection(Configuration conf) throws IOException {
+    // re-create connection with bigtable
+    closeConnection(BigtableSyncTableAccessor.getTargetConnection(this), "target");
+
+    // create target connection config
+    // inherit base config, but override connection based on job args
+    Configuration bigtableConf =
+        getBigtableConfiguration(
+            new Configuration(conf),
+            TARGET_BT_PROJECTID_CONF_KEY,
+            TARGET_BT_INSTANCE_CONF_KEY,
+            TARGET_BT_APP_PROFILE_CONF_KEY,
+            "target");
+    Connection targetConn = BigtableSyncTableAccessor.setTargetConnection(this, bigtableConf);
+    BigtableSyncTableAccessor.setTargetTable(this, targetConn, bigtableConf);
+  }
+
+  /**
+   * close connection
+   *
+   * @param conn
+   * @param sourceOrTarget
+   */
+  private void closeConnection(Connection conn, String sourceOrTarget) {
+    try {
+      if (conn != null) {
+        conn.close();
+      }
+    } catch (IOException ioe) {
+      LOG.debug("error closing temporary " + sourceOrTarget + " connection, " + ioe.getMessage());
+      ioe.printStackTrace();
```

Comment:
why not bubble this up?

Reply:
the thought was that we want to log it, but since this is a connection we're throwing away, we want to proceed with the job even if we observe a failure here.

Reply:
let me know if it would be better if we bubbled this up.

Reply:
I think it should be

Reply:
thanks for confirming, updated to throw the exception up